Topics/Tech_Stacks/CSS-FlexBox.md

```diff
@@ -0,0 +1,124 @@
+### [Conclusion](#conclusion-1)
+
+## Introduction
+
+Flexible Box Layout or often known as FlexBox, is a CSS layout model that allows us to organize our website in a simple yet effective way. Before this layout was introduced to CSS, there were four other layout modes, that includes:
+-	Inline: To organize texts in our webpage
+-	Table: To organize in a 2D-grid based system
+-	Block: To organize grouped or sections of our webpage
+-	Positioned: To specify exact position for items. 
+
+These four layout modes required lots of code to do some simple tasks like “Centering a div”. This is where FlexBox comes in clutch! This not only made coding easier, but also opened CSS Webpage-organization to a whole new level. 
+
+<img width="541" alt="flexbox" src="https://github.com/Raiyankr/CSS-Flexbox/assets/110127056/ebbf9c61-a5db-4f62-bbb7-010acd17009d">
+
+[Image source](https://acciojob.com/blog/untitled-19/)
+
+## How to use FlexBox?
+
+There are two things we need to understand in order to use FlexBox. 
+- Flex Container: This container will contain some items from our webpage. This container will also define the organization property of the items combined.
+- Flex Item: These can be individual components in our webpage (anywhere from a div, to a nested div).
+
+Before we learn about flex containers, lets talk about flex items. These are any component that is nested or "contained" in our flex container. 
+
+```
+<div class="flex-container">
+  <div><1/div>
+  <div>2</div>
+  <div>3</div>
+  <div>4</div>
+</div>
+```
+![flex-item-example](https://github.com/Raiyankr/CSS-Flexbox/assets/110127056/31a39299-5dc9-42a2-b503-c1b9f038012d)
+[Image source](https://www.w3schools.com/css/css3_flexbox_container.asp)
+
+In the above pictures, we have four items in our flex-container. These items will appear in the same order we add them in. It is sufficient to know this much about flex items for now. Let us now learn how we can create a Flex Container. We can do so by using  "_Display: Flex_" property. This simply makes our container *Flex 💪🏻*. 
+
+```
+.flex-container{
+  display: flex;
+}
+```
+
+There are lots of other properties that we can play around with in our container such as:
+- flex-direction
+- flex-wrap
+- justify-content
+- align-items
+
+These properties can be used to change how we want to oprganize our items (vertically or horizontally), whether we want to center our whole container, whether we want to center our items, how do we want to space the items, etc. The following image shows how we can play around with the flex propoerties to do various tasks:
+
+![flex-properties](https://github.com/Raiyankr/CSS-Flexbox/assets/110127056/27c9a05e-7c0d-472a-b3f7-8afc27a80373)
+[Image source](https://www.bitdegree.org/learn/css-flexbox)
+
+To learn more about these properties, [W3Schoole](https://www.w3schools.com/css/css3_flexbox.asp) and [BitDegree](https://www.bitdegree.org/learn/css-flexbox) are great resources!
+
+## Tips and Tricks 
+
```

Comment:
Hey I am not quite sure what you mean by this. Did you mean to move those extra resources under the 'Tips and Tricks' section?

Reply:
having 'Tips and Tricks' as a section header isn't clear as to what content it might have at first glance, and when you look at the content inside, its all about explaining the flex properties you introduced in the previous section. So it might be more appropriate to merge the two sections or make it more explicit what this section is talking about

Reply:
Thank you for clarifying it, I added sub-sections inside this section to make it clear what is it about. Take a look and let me know if this is what you had in mind.

Reply:
Yeah that looks better! its more clear now what each part of the section is talking now!